com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/fs/ResourceUtil.java

```diff
@@ -32,15 +56,109 @@ public static String changeExt(String fileName, String ext) {
     }
 
     /**
-     * Get extension of filename
-     * @param fileName file-name to get extension for
-     * @return the ext, including the '.' character
+     * Optionally change suffix of filename if the requested input suffix matches
+     * @param path path to change suffix for
+     * @param from input suffix
+     * @param to output suffix
+     * @return modified path if input suffix matched, otherwise the original input path
      */
-    public static String getExt(String fileName) {
-        int i = fileName.lastIndexOf(".");
-        if (i == -1) {
-            throw new IllegalArgumentException(String.format("Missing extension in name '%s'", fileName));
+    public static String replaceExt(String path, String from, String to) {
+        if (path.endsWith(from)) {
+            return path.substring(0, path.lastIndexOf(from)).concat(to);
+        }
+        return path;
+    }
+
+   /**
+    * Get the output suffix from an input siffix
+    * @param inExt the input file suffix (including the '.')
+    * @return the output suffix (including the '.')
+    */
+    public static String getOutputExt(String inExt) {
+        String outExt = extensionMapping.get(inExt); // Get the output ext, or use the inExt as default
+        if (outExt != null)
+            return outExt;
+        return inExt;
+    }
+
+    // returns suffix with ".suffix"
+    public static String getSuffix(String path) {
+        int index = path.lastIndexOf(".");
+        if (index < 0)
+            return null;
+        return path.substring(index);
+    }
+
+    public static void enableMinification(boolean enable) {
+        minifyPathEnabled = enable;
+    }
+
+    public static boolean isMinificationEnabled() {
+        return minifyPathEnabled;
+    }
+
+    public static String minifyPathAndReplaceExt(String path, String from, String to) {
+        path = replaceExt(path, from, to);
+        return minifyPath(path);
+    }
+
+    public static String minifyPathAndChangeExt(String path, String to) {
+        path = changeExt(path, to);
+        return minifyPath(path);
+    }
+
+    public static void disableMinify(String ext) {
+        minifyCounters.remove(ext);
+    }
+
+    public static String minifyPath(String path) {
```

Comment:
The main code path that all resource paths go through.
This makes it easier to come up with new naming schemes later. E.g. if we want to use path hashes instead.